src/main/java/de/srendi/advancedperipherals/common/addons/refinedstorage/RSApi.java

```diff
@@ -0,0 +1,782 @@
+        for (TrackedResourceAmount trackedResource : storage.getResources(Actor.EMPTY.getClass())) {
+            if (trackedResource.resourceAmount().resource() instanceof ItemResource itemResource && filter.test(itemResource.toItemStack())) {
+                return getObjectFromItemResource(trackedResource.resourceAmount());
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Returns the first fluid parsed to a lua object which fits to the filter
+     *
+     * @param network refined storage network
+     * @param filter  fluid filter instance - can be an empty filter to get the first fluid of the system see {@link FluidFilter#createEmpty()}
+     * @return the first fluid in the system that fits the fluid filter or null
+     */
+    @Nullable
+    public static Map<String, Object> getParsedFluid(Network network, FluidFilter filter) {
+        StorageNetworkComponent storage = network.getComponent(StorageNetworkComponent.class);
+        for (TrackedResourceAmount trackedResource : storage.getResources(Actor.EMPTY.getClass())) {
+            if (trackedResource.resourceAmount().resource() instanceof FluidResource fluidResource && filter.test(VariantUtil.toFluidStack(fluidResource, trackedResource.resourceAmount().amount()))) {
+                return getObjectFromItemResource(trackedResource.resourceAmount());
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Returns the first mekanism chemical parsed to a lua object which fits to the filter
+     *
+     * @param network refined storage network
+     * @param filter  chemical filter instance - can be an empty filter to get the first chemical of the system see {@link ChemicalFilter#createEmpty()}
+     * @return the first chemical in the system that fits the chemical filter or null
+     */
+    @Nullable
+    public static Map<String, Object> getParsedChemical(Network network, ChemicalFilter filter) {
+        StorageNetworkComponent storage = network.getComponent(StorageNetworkComponent.class);
+        for (TrackedResourceAmount trackedResource : storage.getResources(Actor.EMPTY.getClass())) {
+            if (trackedResource.resourceAmount().resource() instanceof ChemicalResource chemicalResource && filter.test(ChemicalUtil.toChemicalStack(chemicalResource.chemical(), trackedResource.resourceAmount().amount()))) {
+                return getObjectFromChemicalResource(trackedResource.resourceAmount());
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Returns every item from the system while also checking if the filter test passes for the items
+     * The filter can be empty, see {@link ItemFilter#createEmpty()}
+     *
+     * @param network the rs network
+     * @param filter  The filter here is optional, if an empty filter is provided, the method will return every resource
+     * @return a set of items
+     */
+    public static Set<Map<String, Object>> getParsedItems(Network network, ItemFilter filter) {
```

Comment:
I really want a explanation about why are we using `Set` instead of `List` for these results.

Is it because storage may returns multiple same resources? or is it just a personal preference to use `Set` when possible.

Reply:
At the time where I started this, I somehow thought that Sets would be faster for just inserting and then iterating through it. But I was wrong, I researched that back then and Lists were actually faster which makes sense compared to a HashSet implementation. So I moved to Lists for the next functions but I never switched the functions which use Sets to a List - so I will do that next